airbyte-cdk/python/airbyte_cdk/sources/types.py

```diff
@@ -53,12 +53,13 @@ def __ne__(self, other: object) -> bool:
 
 
 class StreamSlice(Mapping[str, Any]):
-    def __init__(self, *, partition: Mapping[str, Any], cursor_slice: Mapping[str, Any]) -> None:
+    def __init__(self, *, partition: Mapping[str, Any], cursor_slice: Mapping[str, Any], last_slice: bool = False) -> None:
```

Comment:
is `last_slice` only useful for the global parent cursor? If yes, can we keep it scoped to the global parent cursor instead of introducing a new field?

The concern is that removing or correcting a field from public interfaces is very difficult so we should be conservative about what we add. This could be done by specializing the type of StreamSlice yielded by the global parent cursor.

Alternatively, if we think the field will be useful for other cases, I think we should promote the `flag_last` method up so the field is always properly set instead of only when using a global parent cursor.

Reply:
Created a new class only for GlobalSubstreamCursor.